### DISCUSSION TITLE
Self-Paced Reading Task with the first word masked

### DISCUSSION THREAD

**Discussion by zepliu** [jspsych/jsPsych#2270] [Q&A] [answered]
Hi, thanks to Josh in the jspsych tutorial for introducing how to create plugin for self-paced reading task. I would like to ask how to modify the codes so that the first word in the sentence is masked in the first place. The display should be something like this:

[------------------
The--------------
---dog-----------
------chased----]

The link below shows the complete jspsych-moving-window code in that tutorial.
https://github.com/jspsych/tutorials/blob/master/moving-window/jspsych-moving-window.js

Thanks for any help in advance.

**Comment by becky-gilbert** [comment, marked as answer]
So I'm not sure if this will work, but try initializing the `current_position` variable ([line 30](https://github.com/jspsych/tutorials/blob/2c556532836de556745842fe44af1f70fba9a84b/moving-window/jspsych-moving-window.js#L30)) to -1 instead of 0.

**Reply by zepliu** [reply]
Thank you Becky. This works but would generate one more RT in the end. For example, there are 5 segments in the sentence 'The dog chased the car' and there'll be 6 RTs, the first one is not the reading time for any segments and should be neglected.

**Reply by becky-gilbert** [reply]
Ok, well one option is to ignore the first RT in the data during the processing/analysis stage. But if you want to prevent that RT from being stored in the trial data, then you can modify the line in the plugin that adds each RT to the data ([line 58](https://github.com/jspsych/tutorials/blob/2c556532836de556745842fe44af1f70fba9a84b/moving-window/jspsych-moving-window.js#L58)) so that the RT is not saved when `current_position` is -1 (i.e. when the whole sentence is masked at the start).
```js
function after_response(response_info){
  if (current_position !== -1) {
    rt.push(response_info.rt);
  }
  // ...
```
Does that solution work for you?

**Reply by zepliu** [reply]
It works, thank you for your kind help!
